### ISSUE TITLE
Support for ZIP64 archives?

### ISSUE THREAD

**Issue by sandersk** [IDPF/epubcheck#334]
Does epubcheck support ZIP64 archives (for EPUBs > 4 GB in size)? I ran a test on a ~14 GB EPUB compressed as what I believe is a valid ZIP64 archive, and got the following error:

````
ERROR: test.epub: I/O error: invalid CEN header (bad signature) 

Check finished with warnings or errors
````

If I'm reading the EPUB 3.0 spec correctly, ZIP64 archives are supported (http://www.idpf.org/epub/30/spec/epub30-ocf.html#sec-zip-container-zipreqs)

"OCF ZIP Containers may use the ZIP64 extensions defined as 'Version 1' in section V, subsection G of the application note at [ZIP APPNOTE] and should use only those extensions when the content requires them. OCF Processors must support the ZIP64 extensions defined as 'Version 1'."

I'm not an expert on the ZIP format, so it's quite possible I've just made a mistake here in how I compressed the EPUB. But figured I'd report in case further investigation was merited.

Thanks,
Sanders

**Comment by vivekbsable** [comment]
I am still getting same issue.

```
java -jar  /opt/apache-tomcat-8.0.0/webapps/epubcheck3-0b2/epubcheck-3.0b2.jar '/data/formats/output/bookstore/20160614053934669237/9781471521430.epub' >> /tmp/12_big_epub.txt
ERROR: /data/formats/output/bookstore/20160614053934669237/9781471521430.epub: I/O error: invalid CEN header (bad signature) 

Check finished with warnings or errors!
```

Java version:
```
[iv2user@develpoment-24 reader]$ java -version
openjdk version "1.8.0_91"
OpenJDK Runtime Environment (build 1.8.0_91-b14)
OpenJDK 64-Bit Server VM (build 25.91-b14, mixed mode)

```
File Size is 2.1 GB

```
[iv2user@develpoment-24 reader]$ du -h /data/formats/output/bookstore/20160614053934669237/9781471521430.epub
2.1G    /data/formats/output/bookstore/20160614053934669237/9781471521430.epub

```

**Comment by arulselvanrpallavarayar** [comment]
Hi, This is due to invalid package/zip file.

Regards, 
 
Arulselvan R Pallavarayar
Direct: +91 (0)44 4916 2259 | Ext: 259
Mobile: +91 (0)9710416093  | arul.selvan@adi-mps.com

**Comment by vivekbsable** [comment]
No, epub contains video and audio files of 2.0 GB. When I remove video and audio file then it is working.


```
[iv2user@develpoment-24 epubcheck4.0.1]$ java -jar  /opt/apache-tomcat-8.0.0/webapps/epubcheck3-0b2/epubcheck-3.0b2.jar '/data/formats/output/bookstore/20160614053934669237/NOID_1.epub' >> /tmp/12_big_epub.txt 
ERROR: /data/formats/output/bookstore/20160614053934669237/NOID_1.epub/OPS/css/NOID.css: invalid operator before a function.

Check finished with warnings or errors!

```

**Comment by vivekbsable** [comment]
> Zip type is ZIP 64+

which is doing something. because when I Copy EPub on **Window OS and did Unzip** and again ZIP, 
after that I checked with ePub Check then it its working.

> 
More important, this time ZIP Type is Zip and (No 64+)

**Comment by rdeltour** [comment]
> No, epub contains video and audio files of 2.0 GB

This looks like an edge case.

Is this still an issue for anyone? I'm considering closing as `wontfix` , unless someone upvotes this issue or provide a repro.

**Comment by murata2makoto** [comment]
@vivekbsable Does your ZIP file correctly announce the use of ZIP 64?  @rdeltour  MS Office can correctly handle OOXML files containing huge video.  Since somebody might want to embed huge video within an EPUB publication, I would like to know more about this issue.
